src/process/mod.rs

```diff
@@ -983,85 +1134,16 @@ impl Process {
 
     /// Return a list of the currently mapped memory regions and their access permissions, based on
     /// the `/proc/pid/maps` file.
-    pub fn maps(&self) -> ProcResult<Vec<MemoryMap>> {
-        let file = FileWrapper::open_at(&self.root, &self.fd, "maps")?;
-
-        let reader = BufReader::new(file);
-
-        let mut vec = Vec::new();
-
-        for line in reader.lines() {
-            let line = line.map_err(|_| ProcError::Incomplete(Some(self.root.join("maps"))))?;
-            vec.push(MemoryMap::from_line(&line)?);
-        }
-
-        Ok(vec)
+    pub fn maps(&self) -> ProcResult<MemoryMaps> {
+        MemoryMaps::from_path(self.root.join("maps"))
```

Comment:
Instead of loading by path, can we use the same `open_at` technique that the old version has?  Something like:

```rust
MemoryMaps::from_reader(FileWrapper::open_at(&self.root, &self.fd, "maps")?)
```

Reply:
I couldn't really see any advantage to using `open_at`, but if you prefer that I can change it.

Reply:
`open_at` helps in case there's pid reuse.  The scenario looks like:

You create a `Process` object for pid `123`.  Internally, we open the `/proc/123` directory.  Some time later, this process exists, but our `Process` object still exists.    If another process is later started that also has pid `123`, then opening `/proc/123/maps` by path will be successful.  But this is problematic because we've now reading maps info for a totally different process.

With `open_at` on the previously opened `/proc/123` directory, trying to open the maps file will result in an error indicating that the process is no longer running (which is what we want)

Reply:
Understood! Made the change.